apps/smart-contracts/token/test/MiniSales.test.ts

```diff
@@ -435,4 +455,180 @@ describe('=> MiniSales', () => {
       await expect(tx).to.emit(miniSales, 'PurchaseHookChange').withArgs(JUNK_ADDRESS)
     })
   })
+
+  describe('# testEndToEnd', () => {
+    let disallowedSourceUser: SignerWithAddress
+    let disallowedDestinationUser: SignerWithAddress
+    let blockedRecipient: SignerWithAddress
+    let ppo: PPO
+    let allowlistPurchaseHook: AllowlistPurchaseHook
+    let sourceAllowlist: AccountList
+    let destinationAllowlist: AccountList
+    let blocklist: AccountList
+    let purchaseAllowlist: AccountList
+    let restrictedTransferHook: RestrictedTransferHook
+    let allowedSourceUser: SignerWithAddress
+    let allowedDestinationUser: SignerWithAddress
+    let blockedUser: SignerWithAddress
+    let unblockedUser1: SignerWithAddress
+    let allowedPurchaser: SignerWithAddress
+
+    const getSignersAndMockTokenForEndToEndStack = async (): Promise<void> => {
+      await setupSignersAndMockTokens()
+      ;[allowedSourceUser, allowedDestinationUser, blockedUser, unblockedUser1, allowedPurchaser] =
+        await ethers.getSigners()
+    }
+
+    const setupEndToEndStack = async (): Promise<void> => {
+      await getSignersAndMockTokenForEndToEndStack()
+      // Setup PPO
+      ppo = await ppoFixture('prePO Token', 'PPO', owner.address)
+      await ppo.connect(owner).acceptOwnership()
+      // Setup MiniSales with PPO as saleToken
+      miniSales = await miniSalesFixture(
+        ppo.address,
+        paymentToken.address,
+        saleTokenDecimals,
+        owner.address
+      )
+      await miniSales.connect(owner).acceptOwnership()
+      // Setup purchase allowlist
+      purchaseAllowlist = await accountListFixture(owner.address)
+      await purchaseAllowlist.connect(owner).acceptOwnership()
+      await purchaseAllowlist
+        .connect(owner)
+        .set([owner.address, allowedPurchaser.address], [true, true])
+      // Setup allow list purchase hook
```

Comment:
```suggestion
      // Setup AllowlistPurchaseHook
```
You are referring directly to the contract that will be used, so capitalization according to the contract name should be used.